Client/Ecosia/Helpers/EcosiaInstallType.swift

```diff
@@ -0,0 +1,85 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/// Represents the type of Ecosia installation.
+enum EcosiaInstallType: String {
+    
+    /// Represents a fresh installation of Ecosia.
+    case fresh
+    
+    /// Represents an upgrade from a previous version of Ecosia.
+    case upgrade
+    
+    /// Represents an unknown installation type.
+    case unknown
+
+    // MARK: - Private Properties
+    
+    /// The key used to store and retrieve the install type from UserDefaults.
+    static let installTypeKey = "installTypeKey"
+    
+    /// The key used to store and retrieve the current installed version from UserDefaults.
+    static let currentInstalledVersionKey = "currentInstalledVersionKey"
+    
+    // MARK: - Public Methods
+    
+    /// Retrieves the current Ecosia install type from UserDefaults.
+    ///
+    /// - Returns: The current Ecosia install type. If not found, returns `.unknown`.
+    static func get() -> EcosiaInstallType {
+        guard let rawValue = UserDefaults.standard.string(forKey: Self.installTypeKey),
+              let type = EcosiaInstallType(rawValue: rawValue)
+        else { return unknown }
+
+        return type
+    }
+
+    /// Sets the Ecosia install type in UserDefaults.
+    ///
+    /// - Parameter type: The Ecosia install type to be set.
+    static func set(type: EcosiaInstallType) {
+        UserDefaults.standard.set(type.rawValue, forKey: Self.installTypeKey)
+    }
+
+    /// Retrieves the persisted current version of Ecosia from UserDefaults.
+    ///
+    /// - Returns: The persisted current version. If not found, returns an empty string.
+    static func persistedCurrentVersion() -> String {
+        guard let currentVersion = UserDefaults.standard.string(forKey: Self.currentInstalledVersionKey) else { return "" }
+
+        return currentVersion
+    }
+
+    /// Updates the persisted current version of Ecosia in UserDefaults.
+    ///
+    /// - Parameter version: The version to be persisted.
+    static func updateCurrentVersion(version: String) {
+        UserDefaults.standard.set(version, forKey: Self.currentInstalledVersionKey)
+    }
+}
+
+extension EcosiaInstallType: Equatable {}
+
+extension EcosiaInstallType {
+    
+    /// Evaluates and updates the current Ecosia install type based on the persisted data and the provided app version.
+    ///
+    /// If the current install type is `.unknown`, it sets the install type to `.fresh` and updates the current version.
+    /// If the persisted version is different from the provided app version, it sets the install type to `.upgrade` and updates the current version.
+    ///
+    /// - Parameter versionProvider: An object conforming to `AppVersionInfoProvider` that provides the current app version.
+    static func evaluateCurrentEcosiaInstallTypeWithVersionProvider(_ versionProvider: AppVersionInfoProvider) {
```

Comment:
Veery minor, but I would suggest adding `DefaultAppVersionInfoProvider()` as the default value here so it doesn't need to be passed when calling from the `AppDelegate`.

Reply:
That's a good point here 😄 . I did make it so the provider was defaulting to the `DefaultAppVersionInfoProvider` but I thought that it may be cleaner that `EcosiaInstallType` would have to only `import Foundation`